lib/npm/routes.js

```diff
@@ -223,14 +245,65 @@ module.exports = function (app) {
       carpenterBuild(opts, done);
     });
   }
+
+  /**
+   * Tags all dependents in a release line, forcing it to either just tag the existing build
+   * and update the build-head or run a full carpenterd build if necessary.
+   * @param {Object} dependents - Listing of dependent package from a release line.
+   * Object keys are package names, values are the versions to be tagged
+   * @param {Object} options - Options object
+   * @param {String} options.env - The environment to perform the rollback in
+   * @param {String} options.log - How to log errors
+   * @param {Function} done - function to execute when complete
+   * @private
+   */
+  function tagDependents(dependents, { tag, log }, done) {
+    if (!dependents) return;
+
+    // Loop the dependents and roll them back to that version as well.
+    const depOpts = Object.entries(dependents)
+      .map(([pkg, version]) => ({
+        pkg,
+        tag,
+        version,
+        log,
+        earlyReturn: false
+      }));
+
+    const config = {
+      retry: { retries: 5, min: 50, max: 10000 },
+      limit: 5,
+      ...app.config.get('dependentBuilds')
+    };
+
+    // setImmediate(function () {
```

Comment:
This can be removed.

Reply:
*sigh* yeah, thought I got all this stuff :(